appar.cabal

```diff
@@ -11,13 +11,17 @@ Cabal-Version:          >= 1.6
 Build-Type:             Simple
 Extra-Source-Files:     README
 library
-  GHC-Options:          -Wall -Wcompat -Wnoncanonical-monadfail-instances
+  GHC-Options:          -Wall
   Exposed-Modules:      Text.Appar.String
                         Text.Appar.ByteString
                         Text.Appar.LazyByteString
   Other-Modules:        Text.Appar.Input
                         Text.Appar.Parser
   Build-Depends:        base >= 4 && < 5, bytestring
+  if impl(ghc >= 8.0)
+    GHC-Options:        -Wcompat -Wnoncanonical-monadfail-instances
```

Comment:
while at it, `-Wnoncanonical-monad-instances` is also useful as it isn't implied by `-Wcompat`